setup/setup_blocks.js

```diff
@@ -1,125 +1,136 @@
 let ProvisionBlocks = (api, userId, key, callback = function() {}, status = function() {}) => {
-  status('Creating new PubNub Function...');
 
-  api.request('post', ['api', 'v1', 'blocks', 'key', key.id, 'block'], {
-    data: {
-      name: 'ChatEngine Function',
-      key_id: key.id
-    }
-  }, (err, response) => {
+  var block = null;
 
-    if (err) {
-      callback('Could not create new PubNub Function. Please contact support@pubnub.com.');
-    }
-
-    let block = response.payload;
-
-    status('Creating new after-presence Event Handler...');
+  status('Creating new PubNub Function...');
 
-    $.get('functions/state-to-kv.js', (code) => {
+  var addSecretKeyToVault = function() {
+    status('Adding Secret Key to Functions Vault...');
+
+    api.request('put', ['api', 'vault', key.subscribe_key, 'key', 'secretKey'], {
+      contentType: 'application/json',
+      data: JSON.stringify({
+        keyName: 'secretKey',
+        key_id: key.id,
+        subscribeKey: key.subscribe_key,
+        value: key.secret_key
+      })
+    }, (err, response) => {
+
+      if (err) {
+        callback('Could not add Secret Key to Functions Vault. Please contact support@pubnub.com');
+        return;
+      }
+
+      status('Success!');
+
+      callback(null, {
+        pub: key.publish_key,
+        sub: key.subscribe_key
+      });
+    });
+  }
+
+  var startPubNubFunction = function() {
```

Comment:
if we are assuming the client is using a modern browser, lets use `let` instead of `var`. Also use `=>` instead of  `function`

Reply:
can't, this is ECMA5.